py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py

```diff
@@ -153,157 +152,80 @@ def native_group_norm(
-        f"{name}_sub",
-        reshaped_input,
-        mean_trt,
-    )
-
-    # variance = mean(pow(sub_trt, 2))
-    pow_trt = get_trt_tensor(ctx, 2, f"{name}_power", np.float32)
-    pow_var = impl.elementwise.pow(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_pow",
-        sub_trt,
-        pow_trt,
-    )
-
-    var_trt = impl.reduce.mean(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_mean_var",
-        pow_var,
-        dim,
-        True,
-    )
-
-    # sqrt((var + eps))
-    eps_trt = get_trt_tensor(ctx, eps, f"{name}_eps", np.float32)
-    add_trt = impl.elementwise.add(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_add",
-        var_trt,
-        eps_trt,
-    )
-    sqrt_trt = impl.unary.sqrt(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_sqrt",
-        add_trt,
-    )
-
-    # y = (X - E[X]) / sqrt((var + eps))
-    div_trt = impl.elementwise.div(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_div",
-        sub_trt,
-        sqrt_trt,
-    )
-
-    # y * gamma + beta
-    gamma_trt = get_trt_tensor(ctx, weight, f"{name}_gamma")
-    beta_trt = get_trt_tensor(ctx, bias, f"{name}_beta")
-
-    output = impl.shuffle.reshape(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_reshape_div",
-        div_trt,
-        input.shape,
-    )
-
-    weight_bias_shape = (1, C) + (1,) * (len(input.shape) - 2)
-
-    reshaped_gamma = impl.shuffle.reshape(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_reshape_gamma",
-        gamma_trt,
-        weight_bias_shape,
-    )
-
-    output = impl.elementwise.mul(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_mul_gamma",
-        output,
-        reshaped_gamma,
-    )
-
-    reshaped_bias = impl.shuffle.reshape(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_reshape_beta",
-        beta_trt,
-        weight_bias_shape,
+        shape,
     )
 
-    output = impl.elementwise.add(
-        ctx,
-        target,
-        source_ir,
-        f"{name}_add_beta",
-        output,
-        reshaped_bias,
+    weight = get_trt_tensor(ctx, weight, f"{name}_weight")
+    bias = get_trt_tensor(ctx, bias, f"{name}_bias")
+    if tuple(reshaped_input.shape) != tuple(weight.shape):
+        weight = impl.slice.expand(
+            ctx,
+            target,
+            source_ir,
+            f"{name}_expand_weight",
+            weight,
+            reshaped_input.shape,
+        )
+    if tuple(reshaped_input.shape) != tuple(bias.shape):
+        bias = impl.slice.expand(
+            ctx, target, source_ir, f"{name}_expand_bias", bias, reshaped_input.shape
+        )
+    dims = list(range(1, len(input.shape)))
+    axes = get_axes_for_reduce_op(dims)
+    group_norm = ctx.net.add_normalization(reshaped_input, weight, bias, axes)
+    group_norm.epsilon = eps
+    group_norm.compute_precision = input.dtype
```

Comment:
Why are we setting this ?

Reply:
@peri044  this was in the layer_norm implementation
https://github.com/pytorch/TensorRT/blob/main/py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py#L126-L129
Since layer_norm and group_norm both uses the INormalizationLayer, so I set it as well.
Also in the INormalizationLayer doc: https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html#inormalizationlayer